src/Extensions/ElementalPageExtension.php

```diff
@@ -98,6 +141,32 @@ public function MetaTags(&$tags)
         }
     }
 
+    private function getEagerLoadedElements(): array
+    {
+        $elements = [];
+        if (is_null(self::$elementalAreas)) {
+            self::$elementalAreas = [];
+            foreach (ElementalArea::get()->eagerLoad('Elements') as $elementalArea) {
+                self::$elementalAreas[$elementalArea->ID] = $elementalArea;
+            }
```

Comment:
It's essentially doing another level of eager loading by getting ALL ElementalArea's at once. It needs to be this way to get any sort of benefit out of what we're doing,

On most websites there is only going to be a single ElementalArea per Page. ElementalPageExtension is an extension applied to Page's. So, in this context, $ids is simply going to be [<$this->owner->ID>]. This means we're going to have many queries to fetch a single ElementalArea instead of a single one

But it gets worse ...  `eagerLoad('Elements')` will only load elements where the parent (in this case ElementArea) ID is in the DataList ... which in this case in only a single ElementalArea, so for my test case it'll only fetch 5 Elements.  So there will subsequently be a large number of queries to the Element table too.